### ISSUE TITLE
Encoding issue with r.http on Windows

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5924]
`r.http` returns very weird results on Windows.

For example `r.http('https://api.github.com/repos/rethinkdb/rethinkdb/stargazers', {resultFormat: "text"})` on Linux returns a reasonable JSON string. But on Windows, it looks like binary data:
![capture2](https://cloud.githubusercontent.com/assets/505365/16671714/ece8ef2c-4455-11e6-8cc1-9576a1f41ebe.PNG)

**Comment by danielmewes** [comment]
The problem seems to be related to gzip or deflate compression.

If I disable it by removing the line
```c++
exc_setopt(curl_handle, CURLOPT_ENCODING, "deflate;q=1, gzip;q=0.5", "PROTOCOLS");
```

from our code, it works fine. It looks like curl doesn't decompress the data on Windows. Maybe it's related to how we compile it?

We could disable compression for `r.http` on Windows until figure this out.

**Comment by danielmewes** [comment]
A fix for this that makes us use only the compression methods that libcurl actually supports on the given system is now in internal code review 3686.

**Comment by danielmewes** [comment]
Fixed in next and v2.3.x. Will ship with RethinkDB 2.3.5.
